frontend/src/Component/History/SensorFilter.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from "react";
+
+// SensorFilter component for selecting device and date range
+export const SensorFilter = ({ onFilterChange }) => {
+  const [beforeDate, setBeforeDate] = useState("");
+  const [afterDate, setAfterDate] = useState("");
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    onFilterChange({ beforeDate, afterDate });
+  };
+
+  return (
+    <form onSubmit={handleSubmit} className="p-4 mb-8 border border-gray-300 rounded-lg shadow-md">
+      <div className="mb-4">
+        <label className="block mb-2 text-sm font-medium text-gray-700">Before Date:</label>
+        <input
+          type="date"
+          value={beforeDate}
+          onChange={(e) => setBeforeDate(e.target.value)}
+          className="w-full p-2 border border-gray-300 rounded"
+          required
+        />
+      </div>
+      <div className="mb-4">
+        <label className="block mb-2 text-sm font-medium text-gray-700">After Date:</label>
+        <input
+          type="date"
+          value={afterDate}
+          onChange={(e) => setAfterDate(e.target.value)}
+          className="w-full p-2 border border-gray-300 rounded"
+          required
+        />
```

Comment:
Could we change this to start date and end date as before date and after date is a bit confusing to figure out which date goes where.

Reply:
I have addressed this issue in the latest commit